test/ingress/aws/aws_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aws
+// +build e2e
+
+package aws
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAWSIngress(t *testing.T) {
+	client := testutil.CreateKubeClient(t)
+
+	i, err := client.ExtensionsV1beta1().Ingresses("httpbin").Get("httpbin", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("getting httpbin ingress: %v", err)
+	}
+
+	h := i.Spec.Rules[0].Host
+
+	resp, err := http.Get(fmt.Sprintf("https://%s/get", h))
+	if err != nil {
+		t.Fatalf("sending HTTP request: %v", err)
+	}
+
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			t.Logf("closing HTTP response body: %v", err)
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("got an error HTTP response: %d", resp.StatusCode)
+	}
```

Comment:
This might fail for various reasons like the pod is not yet started, TLS cert not generated, etc. I would recommend adding some retry logic here.

Using `wait.PollImmediate` is the easiest way to add retry. And we use it everywhere in the code. 

https://github.com/kinvolk/lokomotive/blob/2f92fbe8789dcb49d707a233fe703f017be7721d/test/components/util/util.go#L133

Reply:
Yes, I've considered this of course. I wanted to optimistically start with the simplest code possible and add complexity as things break. But fair enough, I'll add some retry logic.

Reply:
Well, this failed on the first CI run, so :laughing: :shrug: